### PR TITLE
Add compression to configmap storage driver

### PR DESCRIPTION
**Pull request by databus23**
This is a PR to address #1413.

This PR changes the configmap storage driver to compress
the serialized release before storing it as a base64 encoded string.
This change is backward compatible as it handles existing releases
gracefully by skipping the decompression step when the gzip magic
header is not present.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/helm/1420)

### CODE REVIEW

File: pkg/storage/driver/cfgmaps.go [hunk 1]
```diff
@@ -274,6 +289,21 @@ func decodeRelease(data string) (*rspb.Release, error) {
 		return nil, err
 	}
 
+	// For backwards compatibility with releases that were stored before
+	// compression was introduced we skip decompression if the
+	// gzip magic header is not found
+	if bytes.Equal(b[0:3], magicGzip) {
```
Comment (adamreese) [inline pkg/storage/driver/cfgmaps.go]: Very cool!

**Comment by technosophos** [comment]
As soon as the LF CLA gets signed, let me know and I'll merge. Thanks!

**Comment by databus23** [comment]
I signed it!

**Comment by technosophos** [comment]
Merged! Thanks. Are you interested in us keeping you in the loop if any issues arise from this?

**Comment by databus23** [comment]
@technosophos Sure.

**Pull request merged**